src/System.Management.Automation/engine/remoting/fanin/WSManTransportManager.cs

```diff
@@ -2673,42 +2667,25 @@ public void Dispose()
 
                 if (IntPtr.Zero != _handle)
                 {
-#if !CORECLR
-                    // If we initialized with thread impersonation make sure de-initialize is run with same.
-                    WindowsImpersonationContext impersonationContext = null;
+                    int result = 0;
+
+#if !UNIX
+                    // If we initialized with thread impersonation, make sure de-initialize is run with the same.
                     if (_identityToImpersonate != null)
                     {
-                        try
-                        {
-                            _identityToImpersonate.Impersonate();
-                        }
-                        catch (ObjectDisposedException)
-                        {
-                            _handle = IntPtr.Zero;
-                            return;
-                        }
+                        WindowsIdentity.RunImpersonated(
+                            _identityToImpersonate.AccessToken,
+                            () => { result = WSManNativeApi.WSManDeinitialize(_handle, 0); });
```

Comment:
> () => { result = WSManNativeApi.WSManDeinitialize(_handle, 0); }

This can be simplified to `() => result = WSManNativeApi.WSManDeinitialize(_handle, 0)`